x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_logic.ts

```diff
@@ -73,7 +74,7 @@ interface ResultSettingsValues {
   nonTextResultFields: FieldResultSettingObject;
   serverResultFields: ServerFieldResultSettingObject;
   resultFieldsAtDefaultSettings: boolean;
-  resultFieldsEmpty: boolean;
+  isEverythingDisabled: boolean;
```

Comment:
Hm, I kind of like `resultFieldsEmpty` as a var name, curious about the rename 🤔 `isEverythingDisabled` makes me think the inputs are in a disabled state rather than cleared/unchecked/empty. Not a huge deal tho, I'm just being a butt haha

Reply:
I was just looking for a name that more closely matched the user experience, rather than the data. 

When I read that name when I was pulling this over from the old ent-search I didn't understand why we were disabling the "Save" button when the result fields were empty, like it wasn't clear what `resultFieldsEmpty` actually meant in terms of the user experience ... and what it means is that the user has deselected everything. This point is moot though since we have a unit test that explains this now.

The other reason is because it's not necessarily just if the result fields are empty now ... it's also if they are empty or all of the snippet / raw values are false.

So, you have a good point... maybe... `isNothingSelected`, `areAllOptionsUnchecked`, `areAllOptionsDeselected` ... anything like that sound good?

Reply:
On second thought, I'm quite fine with `resultFieldsEmpty` 687927d3d84eec82959d298865e0b6a26ac391f7

Reply:
On second thought, I'm quite fine with `resultFieldsEmpty` 687927d3d84eec82959d298865e0b6a26ac391f7

Reply:
https://github.com/elastic/kibana/pull/97896/commits/e2e727056adeeed5b4b5676751cd623f570e87f8